blueprints/server-capabilitites.md

```diff
@@ -0,0 +1,164 @@
+
+We propose a feature, called "Server Capabilitites" to solve this problem. Servers will have "capabilitites" and Delivery Services will have "Required Capabilitites," and if a server does not have a required capability, then it should not be manually assignable as an Edge, and a Mid must not be added as a parent for that DS in the Edge ATS config.
+
+Initially, this is completely backwards-compatible on upgrade, because initially no Delivery Service will have any Required Capabilities.
+
+This feature will be completely optional. TC operators who don't need Server Capabilities will simply not create them.
+
+Server Capabilities will be arbitrary strings. The ATS project will not "seed" any, and impose as little direction as possible. For example, Server Capabilitites could be Cache types, Server types, Hardware types, ATS versions, Lua support, other ATS plugin support, or any other features an operator needs to route (or not route) on.
+
+
+## Proposed Change
+
+- Servers have Capabilities (i.e.: CACHE_MEMORY, CACHE_DISK )
+
+- Delivery Services have Required Capabilities (i.e.: CACHE_MEMORY, CACHE_DISK)
+
+- When generating Configuration (e.g. ATS parent.config):
+  - If a Mid Server which is otherwise parented to an Edge does not have all Required Capabilities of a Delivery Service, that Mid will not be inserted as a parent for that Delivery Service’s remap and parent rules.
+  - Backward Compatibility is automatic:
+    - Initially, no Delivery Services have Required Capabilities. Hence, everything behaves like it does today.
+    - If an EDGE server does not have all capabilities required by a DS, that server SHOULD NOT be assignable to that DS. 
+    - If an EDGE server is assigned to a DS which requires capabilities that server has, it SHOULD NOT be possible to remove those capabilities from that server.
+
+- Initially, Server Capability names are limited to `[a-Z]`, `[0-9]`, `_`, and `-`. This allows them to be put in most parts of a URI, as well as being the characters of Base64 URL Encoding. We may decide to allow high unicode later. But for now, since we're not sure yet what usage will look like, it's much easier to add characters later without breaking people, than it is to remove them.
+
+### Traffic Portal Impact
+
+- Server Page
+  - New dropdown to add to a text list: Capabilities
+  - List elements have a button to remove the capability
+  - Dropdown/List is just one option - other GUI components may be used
+  - Component should consider that there could be many Capabilities
+
+- Delivery Service Page
+  - New dropdown to add to a text list: Required Capabilities
```

Comment:
I would go this route IF the server capabilities were part of the delivery service object but since they are not I will probably create a UI experience similar to how you assign servers to a delivery service. For example, when viewing a delivery service, you click "More > Manage Required Capabilities" which takes you to:

tp.domain.com/#!/delivery-services/4/required-capabilities

where you can add and remove required server capabilities as needed. Sound ok?